src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesScheduler.java

```diff
@@ -204,15 +206,21 @@ protected CronJob createCronJob(ScheduleRequest scheduleRequest) {
 				schedulerProperties.get("spring.cloud.deployer.kubernetes.cron.expression") :
 				schedulerProperties.get(SchedulerPropertyKeys.CRON_EXPRESSION);
 		Assert.hasText(schedule, "The property spring.cloud.deployer.cron.expression must be defined");
-		
+
 		String concurrencyPolicy = schedulerProperties.get(KUBERNETES_DEPLOYER_CRON_CONCURRENCY_POLICY);
 		// check default server properties
 		if (!StringUtils.hasText(concurrencyPolicy)) {
 			concurrencyPolicy = this.properties.getCron().getConcurrencyPolicy();
 		}
 		if(concurrencyPolicy==null) {
 			concurrencyPolicy = "Allow";
-		} 
+		}
+
+		Integer ttlSecondsAfterFinished = null;
+		String ttlSecondsAfterFinishedString = schedulerProperties.get(KUBERNETES_DEPLOYER_CRON_TTL_SECONDS_AFTER_FINISHED);
+		if (StringUtils.hasText(ttlSecondsAfterFinishedString) && Pattern.matches("^[0-9]+$", ttlSecondsAfterFinishedString)) {
+			ttlSecondsAfterFinished = Integer.parseInt(ttlSecondsAfterFinishedString);
+		}
```

Comment:
We typically check the deployer props then drop down to the default server props when not set (see [concurrencyPolicy](https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/pull/530/files#diff-901dcb4f8a1f2bf100d52f408bc9e357f0d9b0af8375930b7713f8c62ffabadbR211) above). 

Please add the `ttlSecondsAfterFinished` property to the deployer properties `CronConfig` object and then default to that, something like: 
```java
Integer ttlSecondsAfterFinished;
String ttlSecondsAfterFinishedStr = schedulerProperties.get(KUBERNETES_DEPLOYER_CRON_TTL_SECONDS_AFTER_FINISHED);
if (StringUtils.hasText(ttlSecondsAfterFinishedStr)) {
	ttlSecondsAfterFinished = Integer.parseInt(ttlSecondsAfterFinishedStr);
}
else {
	ttlSecondsAfterFinished = this.properties.getCron().getTtlSecondsAfterFinishedString();
}
```

You will also see that I removed the pattern validation. The reason is that It is better to let it fail during parse which will tell the user "You have an invalid property setting". Otherwise it will just skip the parse and the user will wonder why there ttl setting is not taking effect. Eventually, sometime later, after pulling their hair out, they will see that they accidentally set the value to something like `"10'"` when it was supposed to be `"10"`.  😄